docs/export/export.md

```diff
@@ -0,0 +1,466 @@
+  consumer that was built and deployed before the export, e.g.,
+  an inference system that is already deployed when the exporting is done.
+
+(The particular compatibility window lengths are the same that JAX
+[promised for jax2tf](https://github.com/google/jax/blob/main/jax/experimental/jax2tf/README.md#usage-saved-model),
+and are based on [TensorFlow: Forward Compatibility](https://www.tensorflow.org/guide/versions#graph_and_checkpoint_compatibility_when_extending_tensorflow).
+The terminology “backward compatibility” is from the perspective of the consumer,
+e.g., the inference system.)
+
+What **matters is when the exporting and consuming components were built**,
+not the time when the exporting and the compilation happen.
+For external JAX users, it is
+[possible to run JAX and jaxlib at different versions](https://jax.readthedocs.io/en/latest/jep/9419-jax-versioning.html#how-are-jax-and-jaxlib-versioned)
+; what matters is when the jaxlib release was built.
+
+To reduce chances of incompatibility, internal JAX users should:
+  * **rebuild and redeploy consumer systems as frequently as possible**.
+
+and external users should:
+  * run the exporting and consumer systems with the same version of jaxlib, whenever possible, and
+  * export for archival **with the latest released version of jaxlib**.
+
+The compatibility guarantees do not apply if you bypass the `jax.export` APIs
+to obtain the StableHLO code.
+
+Only a subset of custom calls are guaranteed stable and have
+compatibility guarantees ([see list](https://github.com/search?q=repo%3Agoogle%2Fjax%20_CUSTOM_CALL_TARGETS_GUARANTEED_STABLE&type=code)).
+We continuously
+add more custom call targets to the allowed list along with backwards
+compatibility tests. If you try to serialize
+code that invokes other custom call targets you will get an error
+during exporting.
+
+If you want to disable this safety check for a specific custom call,
+e.g., with target `my_target`, you can add
+`export.DisabledSafetyCheck.custom_call("my_target")` to the
+`disabled_checks` parameter of the `export` method,
+as in the following example:
+
+```python
+>>> import jax
+>>> from jax import export
+>>> from jax import lax
+>>> from jax._src.interpreters import mlir
+
+# override the lowering rule for sin to use a custom call `my_new_sin`
+>>> _ = mlir.register_lowering(lax.sin_p, lambda ctx, o: mlir.custom_call("my_new_sin", operands=[o], result_types=[o.type]).results)
+>>> print(jax.jit(lax.sin).lower(1.).compiler_ir())
+module @jit_sin attributes {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32} {
+  func.func public @main(%arg0: tensor<f32> {mhlo.layout_mode = "default"}) -> (tensor<f32> {jax.result_info = "", mhlo.layout_mode = "default"}) {
+    %0 = stablehlo.custom_call @my_new_sin(%arg0) {api_version = 2 : i32} : (tensor<f32>) -> tensor<f32>
+    return %0 : tensor<f32>
+  }
+}
+
+# If we try to export, we get an error
+>>> export.export(jax.jit(lax.sin))(1.)  # doctest: +SKIP
```

Comment:
`# doctest: +IGNORE_EXCEPTION_DETAIL` will run this line, and assert that it raises an error. It's marginally better than `skip` for the sake of keeping documentation up-to-date

Reply:
I agree, I have tried that first, but doctest still fails, at least when I run locally.

Reply:
There were 3 instances of +SKIP in this file. I tried replacing them with IGNORE_EXCEPTION_DETAIL, and it worked for this one on line 206 but results in an error for the other 2: 

e.g., https://github.com/google/jax/actions/runs/9460739515/job/26060095758?pr=21769

Reply:
This indicates that the top line of the expectedexception doesn't match the actual exception

Reply:
The reason it works on line 206 is that you started the exception with `Traceback (most recent call last):`. I think that's the key thing that `IGNORE_EXCEPTION_DETAIL` matches; the rest below that can be anything.

I think it's work replacing these skips, just for the sake of having more doctest coverage for these docs.

Reply:
I agree that is it worth replacing these skips, and I have tried multiple times. All the skipped doctests do raise exceptions and they do start with `Traceback (most recent call last):`

As I was writing this I had the idea to set `JAX_TRACEBACK_FILTERING=off` and with this, I can use `IGNORE_EXCEPTION_DETAIL.